doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,144 @@
+To grow up a protocol method that returns raw JSON to a model, steps are:
+
+**Pick the best grow-up method name fitting your scenario.** You could use the same method name as the initial protocol method only when your grow-up method has different **required** parameter list. E.g.,
+- ```(string feedId, RequestContext requestContext = null)``` and ```(string feedId, CancellationToken cancellationToken = default)``` have the same required parameter list. You cannot add the grow-up method as an overload method by leveraging the initial method name, because it will cause an ambiguous compile error when you call it without `RequestContext`/`CancellationToken`.
+- ```(RequestContent content, RequestContext requestContext = null)``` and ```(string feedId, CancellationToken cancellationToken = default)``` have different required parameter list. You are safe to use same method name with the initial protocol name.
+
+If your grow-up method has the same parameter list as that of the protocol method, we suggest 
+- Adding a suffix `Value` to the initial method name as the grow-up method name, if the initial method is singular (e.g., `GetMetricFeedbackAsync`).
+- Adding a suffix `Values` to the initial method name as the grow-up method name, if the initial method is plural (e.g., `GetMetricFeedbacksAsync`).
+
+**Add implict casting or helper method in generated or handcrafted model.** Examples for model `MetricFeedback` (Models/MetricFeedback/MetricFeedback.cs) are:
+```C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricFeedback
+    {
+        // Add implicit casting
+        public static implicit operator MetricFeedback(Response response)
+        {
+            // Add your deserialization logic in DeserializeMetricFeedback
+            return DeserializeMetricFeedback(JsonDocument.Parse(response.Content.ToMemory()).RootElement);
+        }
+
+        // Add helper method 
+        internal static MetricFeedback FromResponse(Response response)
+        {
+            using var document = JsonDocument.Parse(response.Content);
+            // Add your deserialization logic in DeserializeMetricFeedback
+            return DeserializeMetricFeedback(document.RootElement);
+        }
+    }
+}
+```
+**Call the protocol method and map returned raw response to model.** You could call the protocol method inside your grow-up method and map the returned raw response to model. See different scenarios and corresponding examples below. 
+
+
+## Improve a GET method that returns raw JSON to return a model
+
+**Generated code before (Generated/MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        public virtual async Task<Response> GetMetricFeedbackAsync(Guid feedbackId, RequestContext context = null)
+        {
+        }
+    }
+}
+```
+**Improve the GET method**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
```

Comment:
We should be clear that this is a different file -- any code developers add to the generated client will be deleted when the client is generated.